### DISCUSSION TITLE
can't get anything to work with wine runner

### DISCUSSION THREAD

**Discussion by ExpandingMan** [lutris/lutris#5016] [General]
I feel like I must be doing something obviously wrong here... games run fine for me via steam or bottles, but in lutris, they always seem to fail inexplicably.  Here is an example with Grim Dawn, a game I'm using for testing because I know it runs fine on just about any wine/proton version:
```
lutris-wrapper: Grim Dawn
Started initial process 313661 from /home/expandingman/.local/share/Steam/steamapps/common/Proton - Experimental/files/bin/wine /home/expandingman/Games/steam/Grim Dawn/Grim Dawn.exe
Start monitoring process.
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely

(wine:313757): GStreamer-WARNING **: 12:22:08.511: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstnice.so': /usr/lib/libsoup-3.0.so.0: undefined symbol: nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation
wine: configuration in L"/home/expandingman/Games/steam-prefixes/219990/pfx" has been updated.
Initial process has exited (return code: 13568)
Monitored process exited.
Exit with return code 13568
```
The pattern always seems similar, I always get some high number return code but no further logging.  I always seem to get some kind of `GStreamer` lib warning... I don't think this is unusual from proton in steam, though I haven't confirmed I got exactly these warnings.  Unselecting "prefer system libraries" eliminates most of them, but I still get the one you see above.

Any general advice here of what I can do to troubleshoot?  For some reason when it "installs" it also does not seem to set game options properly.  I have manually set the following in this case
![image](https://github.com/lutris/lutris/assets/11379353/57c579f5-0273-4df4-8b74-043611d7b89c)

Thanks!

**Comment by danieljohnson2** [comment]
I'd try disabling the Lutris runtime; that may be the difference between Lutris and other tools you've tried.

It also seems odd that the game and working directory are not inside the prefix, but while that would stop Lutris from guessing the prefix for you, it seems like it should work.

**Comment by ExpandingMan** [comment]
Disabling the lutris runtime doesn't seem to make any difference.  Can you clarify about the game and the working directory?  The steam prefixes by default do not have the game inside them (or am I crazy?)... I assume they link them when the initialize.

**Comment by ExpandingMan** [comment]
Btw, I'm using the lutris from the manjaro package manager (which is latest).  Perhaps this is a bad idea and I should use flatpak?  Not sure if that's the preferred option.

**Comment by danieljohnson2** [comment]
I don't know how Steam does it, but if you installed the game in Lutris (ie, via an .exe installer stay) Lutris let's the game install inside the prefix.

If you have a Windows game you can install that way, maybe you should try that just to see if it makes any difference. I don't really know why it would, though. Wine can access stuff that is not inside the prefix, but that's where the conventional 'Program Files' directory is.

Flatpak is very likely to be different and may bypass whatever is going wrong, or it may just blow up worse. Another thing to try.

**Comment by ExpandingMan** [comment]
I've tried flatpak and it doesn't seem to make any difference.  Note that I'm simply grabbing the game as it appeared in the "steam" list in lutris.  Perhaps I should try a different way of initializing it?  I've pretty much exhausted all the options on both flatpak and on native, and again I'm not getting anything resembling a useful indication of what might be going wrong, so I'm pretty stumped at this point.

**Comment by danieljohnson2** [comment]
I mean, if you have a game where you have a .exe installer program (ie, not Steam), you can install that. Use the "+" button in the upper left corner to do this.

If you are 100% Steam, maybe that's not a thing you can try.

**Comment by ExpandingMan** [comment]
Yeah I don't have an exe.  So is the assumption that this is missing artifacts?  I don't see why that would be the case since it's a prefix that works when running via steam.

**Comment by danieljohnson2** [comment]
I was thinking there was some sort of incompatibility here, and maybe if everything was set up by Lutris that wouldn't happen.

Just guessing at it, really.

**Comment by ExpandingMan** [comment]
So, more broadly, am I correct in thinking that running steam games with custom runners is not a very good use case for lutris?  Seems likely that whatever is going wrong is relatively simple, but it also seems impossible to resolve without more insight into what's happening.

**Comment by danieljohnson2** [comment]
I couldn't say; not a lot of experience with Steam. But I can't tell what's gone wrong; if you got those warnings with Steam then they are probably not the problem. But we're a little short of clues here.

**Comment by strycore** [comment]
> So, more broadly, am I correct in thinking that running steam games with custom runners is not a very good use case for lutris? 

Why would you even do that. If you want to play Steam games, just use Steam. Lutris is made for non Steam games.

**Comment by ExpandingMan** [comment]
The thing that led me to be experimenting with this was attempting to "hotswap" a different version of `vkd3d`.  As far as I know I'd have to compile a new version of proton to do that on steam, which is probably not that hard once I set it up, so it's on my to-do list.  Anyway, there's clearly some tricksy "gotcha" trying to run steam games on a custom runner in lutris, so it's totally understandable it's not supported, I was just trying to figure out what the situation was.

**Comment by strycore** [comment]
Well it's not only that. Running ANY video game outside of Steam is likely to not work. Steam games depend on the Steam client. There are rare exceptions of games not having Steam DRM but that's really not the norm. If you want to experiment with VKD3D, you have to modify Proton files. There's no way Steam games are going to work with Lutris components.

**Comment by ExpandingMan** [comment]
Fair enough.  I have done something similar before with bottles, but frankly I didn't actually know exactly what bottles was doing.
